source/adapters/opencl/usm.cpp

```diff
@@ -11,6 +11,13 @@
 #include <ur/ur.hpp>
 
 #include "common.hpp"
+#include "usm.hpp"
+
+void AllocDeleterCallback(cl_event event, cl_int, void *pUserData) {
+  clReleaseEvent(event);
+  auto Info = static_cast<AllocDeleterCallbackInfo *>(pUserData);
```

Comment:
No issue but curious why this is `auto` if you know you are casting it to `AllocDeleterCallbackInfo *`?